documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,461 @@
+ClusterManager of my-cluster Group has full access to cluster config on my-cluster
+ClusterManager of my-cluster Group has full access to consumer groups on my-cluster
+ClusterManager of my-cluster Group has full access to topics on my-cluster
+----
+
+`Dev Team A can write to topics that start with x_ on any cluster` combines a resource called `Topic:x_*`, scopes `Describe` and `Write`, and `Dev Team A` policy. The `Dev Team A` policy matches all users that have a realm role called `Dev Team A`.
+
+`Dev Team B can read from topics that start with x_ on any cluster` combines `Topic:x_*`, and `Group:x_*` resources, scopes `Describe` and `Read`, and `Dev Team B` policy. The `Dev Team A` policy matches all users that have a realm role called `Dev Team B`. Matching users and clients have the ability to read from topics, and update the consumed offsets for topics and consumer groups that have names starting with `x_`.
+
+== Targeting permissions using group or role policies
+
+In Keycloak, confidential clients with service accounts enabled can authenticate to the server in their own name using a client ID and a secret.
+This is convenient for microservices which typically act in their own name, and not as agents of a particular user (like a web site would, for example).
+Service accounts can have roles assigned like regular users.
+They cannot, however, have groups assigned.
+As a consequence, if you want to target permissions to microservices using service accounts, you cannot use group policies, and should instead use role policies.
+Conversely, if you want to limit certain permissions only to regular user accounts where authentication with username and password is required, you can achieve that as a side effect of using the group policies, rather than the role policies.
+That is what is used for permissions that start with `ClusterManager`.
+Performing cluster management is usually done interactively - in person - using CLI tools.
+It makes sense to require the user to log in, before using the resulting access token to authenticate to the Kafka Broker.
+In this case, the access token represents the specific user, rather than the client application.
+
+
+== Authorization in action using CLI clients
+
+Make sure that authorization rules have been properly imported.
+
+From menu:Clients[kafka>Authorization>Settings] make sure that *Decision Strategy* is set to *Affirmative*, and NOT to *Unanimous*.
+Navigate in Keycloak to check that the expected resources, authorization claims, policies and permissions are defined.
+
+With the configuration in place, we can check access to Kafka by using a producer and consumer to create topics using different user and service accounts.
+
+First, a new interactive pod container is run using a Strimzi Kafka image to connect to a running Kafka broker.
+
+[source,shell]
+----
+kubectl run -ti --rm --restart=Never --image=quay.io/strimzi/kafka:latest-kafka-2.6.0 kafka-cli -- /bin/sh
```

Comment:
Is this replaced with the right tag at release time? I think it needs to enable replacing variables in the snipper and use some variable which is fixed at release time to create stabel documentation.

Reply:
Do we have a mechanism like that for the docs?

Reply:
There is variable `DockerKafkaImageCurrent` which this. So you need to enable the variable substitution and use something like this:

```
kubectl run -ti --rm --restart=Never --image={DockerKafkaImageCurrent}
```

Reply:
@mstruk Can you look at this so that we can merge this?

Reply:
Ok, here it's just replacing the constants with a variable.